lib/ansible/plugins/lookup/filament_lookup.py

```diff
@@ -0,0 +1,62 @@
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = """
+    lookup: filament_lookup
+    author: Ansible core team
+    version_added: "2.7"
+    short_description: reuturn output of process table.
+    description:
+      - If recieved no arguments, return the whole process table. If recieved arguments, search the process tale with the argument as a process name.
+"""
+
+EXAMPLES = """
+  result: "{{lookup('filament_lookup')}}"
+  tasks:
+  - debug:
+    msg: "result is {{result}}"
+"""
+
+RETURN = """
+content of process table
+"""
+
+
+from ansible.plugins.lookup import LookupBase
+
+try:
+    import subprocess
+    from ansible.errors import AnsibleError
+    from ansible.module_utils._text import to_text
+    import epdb
+    from __main__ import display
+except ImportError:
+    from ansible.utils.display import Display
+    display = Display()
+
+
+def run_command(terms):
+    command = None
+    # if no argument passed, show whole process table
+    if len(terms) is 0:
+        command = "ps aux"
+    # if get an argument, search it in the process table
+    elif len(terms) is 1:
+        command = "ps aux|grep " + str(terms[0])
```

Comment:
Why do you accept a list and then extract a single arg from there anyway? Maybe you could just concatenate all items there?

Reply:
Oh, and in Python we tend to avoid string concatenation with "+"

Reply:
It's because the lookup plugin has no arguments limitation. So the argument the plugin received is a list. And I put the validation code of argument in run_command because it's easy for unit test. So the right way is to have a function do the validation and a function to run command?